crates/kumo-dmarc/src/types/record.rs

```diff
@@ -18,6 +23,73 @@ pub struct Record {
     subdomain_policy: Policy,
 }
 
+impl Record {
+    pub(crate) async fn evaluate(
+        &self,
+        cx: &DmarcContext<'_>,
+        _resolver: &dyn Resolver,
+    ) -> DmarcResultWithContext {
+        match self.align_dkim {
+            Mode::Relaxed => {
+                for dkim in cx.dkim {
+                    if let Some(result) = dkim.get("header.d") {
+                        if !cx.from_domain.ends_with(&format!(".{}", result))
+                            && cx.from_domain != result
+                        {
+                            return DmarcResultWithContext {
+                                result: DmarcResult::Fail,
+                                context: "DMARC: DKIM relaxed check failed".into(),
+                            };
+                        }
+                    }
+                }
+            }
+            Mode::Strict => {
+                for dkim in cx.dkim {
+                    if let Some(result) = dkim.get("header.d") {
+                        if cx.from_domain != result {
+                            return DmarcResultWithContext {
+                                result: DmarcResult::Fail,
+                                context: "DMARC: DKIM strict check failed".into(),
+                            };
+                        }
+                    }
+                }
+            }
```

Comment:
should there be some logic here to handle the "dkim should be strict, but there was no dkim signature" case?  If not, perhaps a comment that says something like `It is OK for there to be no dkim signatures when in strict mode` would be good as a sign post of sorts here?  I can't see an obvious statement about missing dkim signature and its effect on the alighment in the RFC, so I'm not sure which way this should go!

Would be good to have a test for this case if you don't already (I'm currently still in my first pass through, so I don't yet know if that exists!)